### ISSUE TITLE
Section header translations are unnecessarily required to be in title case

### ISSUE THREAD

**Issue by MoreFamed** [robotframework/robotframework#4411]
I am testing the Czech localization. It works just partially for me. I cannot use Czech variants of *** Keyword ***, *** Keywords ***, *** Test Case *** and *** Test Cases *** because of the error `Unrecognized section header '*** Testovací případy ***'. Valid sections: 'Settings', 'Variables',` etc. The other ones I tested (12 other items) seem to be OK when run with `robot --language CS`. I noticed that those problematic ones are those that should be written into asterisks and together with it they consist of more than one word. Document at https://robotframework.crowdin.com/translate/ seems to be good.

RF 5.1 alpha 1

**Comment by pekkaklarck** [comment]
The reason is that these headers are specified in the language definition like

```python
test_case_headers = {'Testovací případ', 'Testovací případy', 'Testovací případy', 'Testovací případy'}
```

and header matching code converts the actual header to title case before comparison. This same problem was actually already reported by @HelioGuilherme66 in PR #4396 and I thought I fixed it in d20dd1c2208e0d539c1e5a075bb4074c14ac30ba. The problem is that I only fixed it with setting names, not with header names. In other words, the problem Hélio reported wasn't actually fixed either.

I'll fix the problem with headers soon and probably create alpha 2 later this week. We already have gotten some new translations and would be nice to get them out as well. Also BDD prefixes seem to have this issue so I'll fix that as well.

Some questions:
- Do you @MoreFamed prefer format `Testovací případy` over `Testovací Případy` in general? After my fix Robot will accept both values in headers, but I expect that the values we use in our language definitions will be later used by editors and other external tools. With English we've decided to prefer the title cased format but different languages have different preferences. Whatever is the preferred format should be used in the language definition.
- As you can see, the language definition currently contains `Testovací případy` three times and it seems that other headers also have duplicates. I assume removing the dupes is fine. Should we @leeuwe possibly enhance `crowdin.py` script to avoid such duplicates?
- Should we start requiring tests for translations? They would have found this bug but I*m not certain how many similar bugs there are anymore. Adding and updating tests for all languages is some work.

**Comment by pekkaklarck** [comment]
This bug affected French, Czech and Portuguese but ought to be fixed now.

I also removed the duplicated Czech header translations. No other language seemed to have dupes.
